### PR TITLE
FIX issues in GET /v2/type and GET /v2/type/{id} operations

### LINKED ISSUE #2046: in GET v2/types does not returns the types total in the "x-total-count" response header
in GET v2/types does not returns the types total in the "x-total-count" response header

With 4 types defined
```
 home type with  5 entities
 none type with  5 entities
 car type with  10 entities
 house type with 5 entities
```
But aplying a limit equals to 2, is returned x-total-count: 2

#### types request
```
GET http://localhost:1026/v2/types?limit=2&options=count&offset=1
Content-Type: application/json
Fiware-Service: test_type_happy_path
Fiware-ServicePath: /test
```

#### types response
```
http code: 200
content-length: 206
x-total-count: 2
fiware-correlator: 61c4ccd0-053b-11e6-b308-005056a20feb
connection: Keep-Alive
date: Mon, 18 Apr 2016 07:58:48 GMT
content-type: application/json
 [{"type":"home","attrs":{"temperature_0":{"types":["none"]},"temperature_1":{"types":["none"]}},"count":5},{"type":"house","attrs":{"pressure_0":{"types":["bar"]},"pressure_1":{"types":["bar"]}},"count":5}]
```

#### expected response
```
...
x-total-count: 4
...
```

### PR DESCRIPTION
**Pull request by fgalan**
Fixes #2046 and #2056 

@crbrox 
@kzangeli

### CODE REVIEW

File: src/lib/mongoBackend/mongoQueryTypes.cpp [hunk 1]
```diff
@@ -157,7 +157,8 @@ HttpStatusCode mongoEntityTypes
   const std::string&                   tenant,
   const std::vector<std::string>&      servicePathV,
   std::map<std::string, std::string>&  uriParams,
-  const std::string&                   apiVersion
+  const std::string&                   apiVersion,
+  unsigned int*                        totalTypes
```
Comment (kzangeli) [inline src/lib/mongoBackend/mongoQueryTypes.cpp]: To indicate that totalTypes is a pointer (and that might get modified inside the function), we usually add a 'P' at the end of the variable name - totalTypesP.

This we should also add to the style guide, as SHOULD.

**Comment by crbrox** [comment]
LGTM

File: src/lib/mongoBackend/mongoQueryTypes.cpp (hunk 1 above)
Comment (fgalan) [inline src/lib/mongoBackend/mongoQueryTypes.cpp]: Fixed in ccc6eca

**Comment by kzangeli** [comment]
Following the link to the commit ccc6eca, src/lib/ngsi/ContextAttribute.cpp contains git diff stuff ...
Not visible under 'Files Changed' of the PR though ...

**Comment by fgalan** [comment]
It seems that a not-so-clean conflict resolution make them appear (my fault). However, I fixed in that commit. Result is that at then end (i.e. diff tab) everythying seems to be ok.

**Comment by kzangeli** [comment]
LGTM

**Pull request merged**